### PR TITLE
Add check for daq-reader processing threads

### PR DESCRIPTION
**Pull request by jorana**
**What is the problem / what does the code in this PR do**
Check that redax / dispatcher is not writing wrong threads to wrong runs.

**Can you briefly describe how it works?**
Each of the readers is registered, let's check that we don't get reader_6 for non NV data, etc.

### CODE REVIEW

**Review by WenzDaniel — APPROVED** [review]
Thank you for the quick fix. Lets hope we wont need it in the future.

**Review by darrylmasson — COMMENTED** [review]
"Sloppy" is a word to describe this. We're passing through separately the keys and values of `daq_config.processing_threads`, and not using them in a combined and coherent way.

**Comment by jorana** [comment]
> 
> 
> "Sloppy" is a word to describe this. We're passing through separately the keys and values of `daq_config.processing_threads`, and not using them in a combined and coherent way.

@darrylmasson I wasn't anticipating that we'd keep this implemented for long, you want me to change it?

**Comment by darrylmasson** [comment]
We have two options: a quick and dirty implementation, or something that makes fuller use of the information we can access. Do we want DAQReader to always just take whatever files are present, or do we think it's reasonable to actually check that it's seeing the files it should? We had a similar problem much earlier when redax wasn't correctly allocating processing threads, and this is quite similar.

**Comment by jorana** [comment]
> 
> 
> We have two options: a quick and dirty implementation, or something that makes fuller use of the information we can access. Do we want DAQReader to always just take whatever files are present, or do we think it's reasonable to actually check that it's seeing the files it should? We had a similar problem much earlier when redax wasn't correctly allocating processing threads

Okay, based on your phrasing, I think you imply the latter. Thanks for the feedback. To implement the latter:
- Indeed just use the dict instead of keys and values separately;
- use a counter to see that the number in each chunk is `le` the #processing threads from the rundoc (last few chunks don't need to have all threads per se).

File: straxen/plugins/daqreader.py [hunk 1]
```diff
@@ -141,12 +143,32 @@ def _chunk_paths(self, chunk_i):
                 result.append(False)
         return tuple(result)
 
+    @staticmethod
+    def _partial_chunk_to_thread_name(partial_chunk):
+        """Convert name of part of the chunk to the thread_name that wrote it"""
+        return '_'.join(partial_chunk.split('_')[-1])
```
Comment (darrylmasson) [inline straxen/plugins/daqreader.py:149]: `reader0_reader_0_threadid` -> `t_h_r_e_a_d_i_d`, is that what we are interested in, or should it be `[:-1]`?

**Review by darrylmasson — CHANGES_REQUESTED** [review]
Some typos

File: straxen/plugins/daqreader.py [hunk 2]
```diff
@@ -101,6 +102,7 @@ def infer_dtype(self):
     def setup(self):
         self.t0 = int(self.config['run_start_time']) * int(1e9)
         self.dt_max = self.config['max_digitizer_sampling_time']
+        self.n_readout_treads = sum(self.config['readout_threads'].values())
```
[the remark this answers is not in the document]
Comment (jorana) [reply straxen/plugins/daqreader.py]: Thanks Darryl, well spotted, should be fixed now.

**Comment by jorana** [comment]
Thanks @darrylmasson, well spotted mistakes. Just tried it on two runs in non production mode. One processed all right the other gave:
`/live_data/xenonnt/012118/THE_END. Got reader6_reader_0`

Are you happy with this behavior with the counting of threads or would you argue for smarter checks?

**Review by darrylmasson — APPROVED** [review]

**Pull request merged**